src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs

```diff
@@ -28,17 +35,142 @@ internal abstract class AbstractInheritanceMarginService : IInheritanceMarginSer
+        {
+            using var _ = ArrayBuilder<InheritanceMarginItem>.GetInstance(out var items);
+
+            await AddInheritedImportsAsync(document, spanToSearch, items, cancellationToken).ConfigureAwait(false);
+            await AddInheritanceMemberItemsAsync(document, spanToSearch, items, cancellationToken).ConfigureAwait(false);
+
+            return items.ToImmutable();
+        }
+
+        private async Task AddInheritedImportsAsync(
+            Document document,
+            TextSpan spanToSearch,
+            ArrayBuilder<InheritanceMarginItem> items,
+            CancellationToken cancellationToken)
         {
             var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var allDeclarationNodes = GetMembers(root.DescendantNodes(spanToSearch));
-            if (allDeclarationNodes.IsEmpty)
+
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var imports = syntaxFacts.GetImportsOfCompilationUnit(root);
+
+            // Place the imports item on the start of the first import in the file.  Or, if there is no import, then on
+            // the first line.
+            var spanStart = imports.Count > 0 ? imports[0].SpanStart : 0;
+
+            // if that location doesn't intersect with the lines of interest, immediately bail out.
+            if (!spanToSearch.IntersectsWith(spanStart))
+                return;
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var scopes = semanticModel.GetImportScopes(root.FullSpan.End, cancellationToken);
+
+            // If we have global imports they would only be in the last scope in the scopes array.  All other scopes
+            // correspond to inner scopes for either the compilation unit or namespace.
+            var lastScope = scopes.LastOrDefault();
+            if (lastScope == null)
+                return;
+
+            // Pull in any project level imports, or imports from other files (e.g. global usings).
+            var syntaxTree = semanticModel.SyntaxTree;
+            var nonLocalImports = lastScope.Imports
+                .WhereAsArray(i => i.DeclaringSyntaxReference?.SyntaxTree != syntaxTree)
+                .Sort((i1, i2) =>
+                {
+                    return (i1.DeclaringSyntaxReference, i2.DeclaringSyntaxReference) switch
+                    {
+                        // Both are project level imports.  Sort by name of symbol imported.
+                        (null, null) => i1.NamespaceOrType.ToDisplayString().CompareTo(i2.NamespaceOrType.ToDisplayString()),
+                        // project level imports come first.
+                        (null, not null) => -1,
+                        (not null, null) => 1,
+                        // both are from different files.  Sort by file path first, then location in file if same file path.
```

Comment:
I wonder if this distinction is useful, or if just sorting them all alphabetically is just as good?

Related: By "project level" do you mean the VB global imports that are specified on the command line? Because if so, isn't it impossible for "project level imports" and "global usings" to appear int he same list, since they're relevant to different languages?

Reply:
> I wonder if this distinction is useful, or if just sorting them all alphabetically is just as good?

I def like sorting as per the items in a file, as that makes it feel like you're getting a little view into that file :)

Reply:
> Related: By "project level" do you mean the VB global imports that are specified on the command line? Because if so, isn't it impossible for "project level imports" and "global usings" to appear int he same list, since they're relevant to different languages?

you are correct.  the above is a little bit of paranoia... i'll see if i can comfortably clean it up though!